sklearn/utils/class_weight.py

```diff
@@ -41,12 +41,17 @@ def compute_class_weight(class_weight, classes, y):
     # Import error caused by circular imports.
     from ..preprocessing import LabelEncoder
 
+    if y.dtype == np.float32:
+        _dtype = np.float32
```

Comment:
why not `_dtype=y.dtype`...

is it so you can have `y.dtype` to be `int` and `weight` to be of `float`?